src/provider/rpc.ts

```diff
@@ -176,7 +178,7 @@ export class RpcProvider implements ProviderInterface {
     // can't be named simulateTransaction because of argument conflict with account
     return this.channel
       .simulateTransaction(invocations, options)
-      .then(this.responseParser.parseSimulateTransactionResponse);
+      .then((r) => this.responseParser.parseSimulateTransactionResponse(r));
```

Comment:
Any specific reason for changing to implicit providing params ?

Reply:
With the changes in this PR `RPCResponseParser` has internal `this` references in the parsing methods so their usage needed to be adapted for the proper `this` to be accessible and changing it to arrow functions was shorter than doing `.bind(this.responseParser)`.